pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -199,7 +199,8 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 		if minUpgradableVersion == "" {
 			logrus.Debug("No minimum upgradable version specified, continue the upgrading")
 		} else {
-			constraint := fmt.Sprintf(">= %s", minUpgradableVersion)
+			// to enable comparisons against prerelease versions
+			constraint := fmt.Sprintf(">= %s-z", minUpgradableVersion)
```

Comment:
prefer to set the minUpgradableVersion to `1.2.3-0` instead according to https://github.com/Masterminds/semver#working-with-prerelease-versions
Also, would be helpful to wrap the comparison check into a method and add unit test for it, thanks.

Reply:
Updated with unit tests. Please take a look, thanks!